fsspec/tests/abstract/__init__.py

```diff
@@ -61,7 +85,7 @@ def local_target(self, local_fs, local_join, local_path):
         if local_fs.exists(target):
             local_fs.rm(target, recursive=True)
 
-    def _bulk_operations_scenario_0(self, some_fs, some_join, some_path):
+    def _bulk_operations_scenario_0(self, some_fs, some_join, some_path, glob=False):
```

Comment:
The initial idea here was that there would be one and only one scenario for all the `copy`-like tests and the tests would be identical to the examples in the docs so that it is easy to demonstrate that the tests are appropriate and easy to follow for future contributors. Having two variants of the scenario 0 breaks this, which makes me slightly uncomfortable, but I can see that it is better this way than using the more complicated `glob` scenario for all the `copy` tests. So I am OK to go ahead with this, but can you add a new diagram in the docstring below to illustrate what the `glob=True` scenario consists of?

Reply:
You're totally right. I reverted my changes of removing the `test_*_glob_to_*_directory` old tests and the `glob: bool` option here. This way it is clearer that globs supports the scenario in the documentation. For the remaining edge cases glob tests, I moved them into a new `test_*_glob_edge_cases` and add a new fixture to create the files for those edge cases.

In the process, I realized that the following assertion was wrong and fixed the files/folder removal in some tests with for loops:
```python
# wrong assertion since it test ` fs.ls(target) == []` in one case and `[dummy]` in the other which is always True
assert fs.ls(target) == [] if supports_empty_directories else [dummy]

# replaced by the following which test ` fs.ls(target) == []` in one case and `fs.ls(target) == [dummy]` in the other
assert fs.ls(target) == ([] if supports_empty_directories else [dummy])
```

Reply:
Thanks.